src/components/common/Modal.jsx

```diff
@@ -0,0 +1,41 @@
+import { createPortal } from 'react-dom';
+
+/* 사용방법
+1. 모달을 사용하고 싶은 곳에 import한다.
+2. <Modal>콘텐츠 내용</Modal> 을 입력한다.
+3. useModal() 훅을 사용한다.
+4. useModal()을 destructing해서 {isOpen, open, close}를 가져온다. {isOpen, open, close} = useModal();
+5. 모달 컴포넌트에 isOpen, close를 Props로 내려준다. isOpen={isOpen}, onClose(close)
+6. open은 모달을 열고 싶은 버튼에 onClick 이벤트로 넣어준다.
+7. 추가로 props로 title, button, extra를 보내줄 수 있다.
+*/
+
+const Modal = ({ title, button, children, extra, isOpen, onClose }) => {
+  if (!isOpen) return;
+
+  return createPortal(
+    <div className='modal-overlay fixed top-0 right-0 bottom-0 left-0 flex h-screen w-screen items-center justify-center bg-black/40'>
+      <div className='modal-content relative flex max-h-8/10 w-9/10 max-w-md min-w-64 flex-col rounded-2xl bg-black p-10 text-white'>

```

Comment:
`max-h-8/10`, `w-9/10`은 사양 상 비표준 클래스 같은데, `[]` 안에 구체적인 값을 넣는 것이 좋을 수도 있을 것 같습니다.

```suggestion
      <div className='modal-content relative flex max-h-[80%] w-[90%] max-w-md min-w-64 flex-col rounded-2xl bg-black p-10 text-white'>
```